tests/migration/reboot_to_upgrade.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Reboot machine to perform upgrade
```

Comment:
For me module should do something then just set two variables and trigger reboot , it's look like a function which will be called inside install_and_reboot module

Reply:
Yes, it's really a simple test module without much function, also true for post_upgrade.pm.
As to install_and_reboot module, I think you meant tests/x11/reboot_and_install.pm
The purpose of this commit is actually the removal of reboot_and_install module from upgrade testing, based on following reasons:
   * reboot_and_install module need switch back to x11 console and reboot from gnome, which is more likely to fail compared to text mode reboot. Several x11 reboot failures had been observed in upgrade testing on o.s.d, and most failures were not real product issue but just caused due to performance problem on over-loaded openqa worker. So, replace x11 reboot with text mode reboot to stabilize the upgrade tests.
   * the codes for rebooting, in reboot_and_install module, look some bit of duplication to existing test modules included in load_boot_tests. And we can catch up with latest update on booting functions to reuse existing test modules.
   * split reboot and install will help adapt version switching better in upgrade test: before reboot, it is original system to patch; after reboot, it is target version to upgrade. Just trigger reboot and do version switch immediately, it will reload needles for new version, which would help for needle match during machine booting.

Reply:
make sense